spring-session/src/main/java/org/springframework/session/data/couchbase/CouchbaseDao.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.session.data.couchbase;
+
+import java.util.List;
+import java.util.Map;
+
+import com.couchbase.client.java.document.json.JsonArray;
+import com.couchbase.client.java.document.json.JsonObject;
+import com.couchbase.client.java.error.DocumentDoesNotExistException;
+import com.couchbase.client.java.query.N1qlQuery;
+import com.couchbase.client.java.query.N1qlQueryResult;
+import com.couchbase.client.java.query.N1qlQueryRow;
+
+import org.springframework.data.couchbase.core.CouchbaseQueryExecutionException;
+import org.springframework.data.couchbase.core.CouchbaseTemplate;
+import org.springframework.util.Assert;
+
+/**
+ * Data access object that communicates with Couchbase database in order to manage HTTP
+ * session data persistence.
+ *
+ * @author Mariusz Kopylec
+ * @since 1.2.0
+ */
+public class CouchbaseDao {
```

Comment:
Is there a reason we need a distinct `CouchbaseDao`? Perhaps this would be better in the `CouchbaseSessionRepository`